builtin/logical/pki/path_config_ca.go

```diff
@@ -13,24 +16,443 @@ func pathConfigCA(b *backend) *framework.Path {
+		PrivateKeyType: csrb.PrivateKeyType,
+	}
+
+	entry, err := logical.StorageEntryJSON("config/ca_bundle", cb)
+	if err != nil {
+		return nil, err
+	}
+	err = req.Storage.Put(entry)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (b *backend) pathCASignIntermediate(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	var err error
+
+	format := data.Get("format").(string)
+	switch format {
+	case "pem":
+	case "der":
+	default:
+		return logical.ErrorResponse(
+			`The "format" path parameter must be "pem" or "der"`,
+		), nil
+	}
+
+	role := &roleEntry{
+		TTL:              data.Get("ttl").(string),
+		AllowLocalhost:   true,
+		AllowAnyName:     true,
+		AllowIPSANs:      true,
+		EnforceHostnames: false,
+	}
+
+	if cn := data.Get("common_name").(string); len(cn) == 0 {
+		role.UseCSRCommonName = true
+	}
+
+	var caErr error
+	signingBundle, caErr := fetchCAInfo(req)
+	switch caErr.(type) {
+	case certutil.UserError:
+		return nil, certutil.UserError{Err: fmt.Sprintf(
+			"could not fetch the CA certificate (was one set?): %s", caErr)}
+	case certutil.InternalError:
+		return nil, certutil.InternalError{Err: fmt.Sprintf(
+			"error fetching CA certificate: %s", caErr)}
+	}
+
+	useCSRValues := data.Get("use_csr_values").(bool)
+
+	maxPathLengthIface, ok := data.GetOk("max_path_length")
+	if ok {
+		maxPathLength := maxPathLengthIface.(int)
+		role.MaxPathLength = &maxPathLength
+	}
+
+	parsedBundle, err := signCert(b, role, signingBundle, true, useCSRValues, req, data)
+	if err != nil {
+		switch err.(type) {
+		case certutil.UserError:
+			return logical.ErrorResponse(err.Error()), nil
+		case certutil.InternalError:
+			return nil, err
+		}
+	}
+
+	cb, err := parsedBundle.ToCertBundle()
+	if err != nil {
+		return nil, fmt.Errorf("Error converting raw cert bundle to cert bundle: %s", err)
+	}
+
+	resp := b.Secret(SecretCertsType).Response(
+		map[string]interface{}{
+			"expiration":    int64(parsedBundle.Certificate.NotAfter.Unix()),
+			"serial_number": cb.SerialNumber,
 		},
+		map[string]interface{}{
+			"serial_number": cb.SerialNumber,
+		})
 
-		HelpSynopsis:    pathConfigCAHelpSyn,
-		HelpDescription: pathConfigCAHelpDesc,
+	switch format {
+	case "pem":
+		resp.Data["certificate"] = cb.Certificate
+		resp.Data["issuing_ca"] = cb.IssuingCA
+	case "der":
+		resp.Data["certificate"] = base64.StdEncoding.EncodeToString(parsedBundle.CertificateBytes)
+		resp.Data["issuing_ca"] = base64.StdEncoding.EncodeToString(parsedBundle.IssuingCABytes)
 	}
+
+	resp.Secret.TTL = parsedBundle.Certificate.NotAfter.Sub(time.Now())
+
+	err = req.Storage.Put(&logical.StorageEntry{
+		Key:   "certs/" + cb.SerialNumber,
+		Value: parsedBundle.CertificateBytes,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Unable to store certificate locally")
+	}
+
+	if parsedBundle.Certificate.MaxPathLen == 0 {
+		resp.AddWarning("Max path length of the signed certificate is zero")
```

Comment:
This is fine if the certificate is only being used to sign end certificates right?